### PR TITLE
SeaDrop x Art Blocks Shim

### PR DESCRIPTION
**Pull request by ryley-o**
## Description of the change

Add shim layer to enable SeaDrop integration for Art Blocks v3.2 core contracts.

The implementation is compatible with the shared minter suite, but one contract must be used for each project using the integration.

closes https://linear.app/art-blocks/issue/PLT-631/production-ready-contracts-after-openseas-review

## Tests

Tests are added for coverage of AB-developed logic.

In an effort to avoid large mocks as part of testing that would add little-to-no value to our test suite, portions of logic copied directly from OpenSea's example implementation of `multiConfigure`, as well as specific calldata forwarded to the SeaDrop implementation, are not directly tested. AB-developed logic remains thoroughly tested as usual for our contracts.

## TODO

- [x] initial design review/edits (AB)
- [x] OpenSea review/feedback
- [x] Finalize implementation + tests

### CODE REVIEW

**Comment by ryley-o** [comment]
@lindsgil requested initial cut at an internal design review - feel free to make major edits/updates based on anything you find!

File: packages/contracts/contracts/interfaces/v0.8.x/ISeaDropShimForContract.sol [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+
+// @dev file adds event for indicating a shim contract compatible with SeaDrop, and points to the corresponding NFT
+// contract
+pragma solidity ^0.8.22;
+
+interface ISeaDropShimForContract {
+    /**
+     * @dev Emit event when a shim-minter compatible with SeaDrop that mints tokens on a different contract is deployed
+     */
+    event SeaDropShimForContract(address nftContract);
```
Comment (lindsgil) [inline packages/contracts/contracts/interfaces/v0.8.x/ISeaDropShimForContract.sol:11]: 🥳

**Comment by lindsgil** [comment]
Looking great so far! 🔥

**Review by yoshiwarab — APPROVED** [review]
🚀🚀🚀

**Pull request merged**